src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -94,9 +96,9 @@ public void execute(JobExecutionContext jobExecutionContext) {
    * @return The saved notification history.
    */
   private History saveNotificationSent(JobDataMap jobDetails, String email, Instant sentAt) {
-    ObjectId objectId = ObjectId.get(); //TODO let DB generate this?
+    ObjectId objectId = ObjectId.get();
     RecipientInfo recipientInfo
-        = new RecipientInfo(jobDetails.getString(TIS_ID), MessageType.EMAIL, email);
+        = new RecipientInfo(jobDetails.getString(PERSON_ID_FIELD), MessageType.EMAIL, email);
```

Comment:
Using the programme membership ID was incorrect here - meaning that old history items were never being found when deciding whether to schedule new notifications, since the persons ID was expected when searching for these.